src/client/datascience/notebook/vscodeNotebookController.ts

```diff
@@ -118,7 +118,6 @@ export class VSCodeNotebookController implements Disposable {
         disposableRegistry: IDisposableRegistry,
         private readonly languageService: NotebookCellLanguageService,
         private readonly workspace: IWorkspaceService,
-        private readonly localOrRemoteKernel: 'local' | 'remote',
```

Comment:
Yay, no need, whether it remote or local, is derived from the kernel connection metadata